### PR TITLE
Release 1.1.0

### PR DESCRIPTION
**Pull request by zippy1978**
This PR prepares release 1.1.0 of the plugin.

- [x] Manual testing with analysis
- [x] Version bump
- [x] Upgrade Dockerfile with SonarQube 10.0
- [x] Fixed rule html description layout
- [x] Updated documentation (README, CHANGELOG, CONTRIBUTING)
- [x] Updated dependencies to address security alerts

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,12 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [1.1.0]
```
Comment (dedece35) [inline CHANGELOG.md:8]: for me, this is the releasing process that will do this king of modifications.
for me, keep "Unreleased" here

File: commons-ios/pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>io.ecocode</groupId>
         <artifactId>ios-plugin</artifactId>
-        <version>1.1.0-SNAPSHOT</version>
+        <version>1.1.0</version>
```
Comment (dedece35) [inline commons-ios/pom.xml:7]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

File: docker-compose.yml [hunk 3]
```diff
@@ -18,8 +18,8 @@ services:
       SONAR_ES_BOOTSTRAP_CHECKS_DISABLE: 'true'
     volumes:
       - type: bind
-        source: ./sonar-ios-plugin/target/ecocode-sonar-ios-plugin-1.1.0-SNAPSHOT.jar
-        target: /opt/sonarqube/extensions/plugins/ecocode-sonar-ios-plugin-1.1.0-SNAPSHOT.jar
+        source: ./sonar-ios-plugin/target/ecocode-sonar-ios-plugin-1.1.0.jar
```
Comment (dedece35) [inline docker-compose.yml:21]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

File: docker-compose.yml [hunk 4]
```diff
@@ -18,8 +18,8 @@ services:
...
+        target: /opt/sonarqube/extensions/plugins/ecocode-sonar-ios-plugin-1.1.0.jar
```
Comment (dedece35) [inline docker-compose.yml:22]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

File: pom.xml [hunk 5]
```diff
@@ -4,7 +4,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.ecocode</groupId>
     <artifactId>ios-plugin</artifactId>
-    <version>1.1.0-SNAPSHOT</version>
+    <version>1.1.0</version>
```
Comment (dedece35) [inline pom.xml:7]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

File: sonar-ios-plugin/pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>io.ecocode</groupId>
         <artifactId>ios-plugin</artifactId>
-        <version>1.1.0-SNAPSHOT</version>
+        <version>1.1.0</version>
```
Comment (dedece35) [inline sonar-ios-plugin/pom.xml:7]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

File: swift-lang/pom.xml [hunk 7]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>io.ecocode</groupId>
         <artifactId>ios-plugin</artifactId>
-        <version>1.1.0-SNAPSHOT</version>
+        <version>1.1.0</version>
```
Comment (dedece35) [inline swift-lang/pom.xml:7]: for me, this is the releasing process that will do this king of modifications.
for me, keep "SNAPSHOT" version here and check if releasing process is OK with ios release (or upgrade it)

**Review by dedece35 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zippy1978) [reply CHANGELOG.md:8]: Ok, done

**Review by dedece35 — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (jycr) [reply pom.xml]: By using the "[Maven Friendly Version](https://maven.apache.org/maven-ci-friendly.html)" guide, there is no more file modification needed in Git for a release.

Comment (dedece35) [reply pom.xml]: @jycr I didn't know this maven feature ! cool feature indeed ! ... but we have to refactor our pom.xml files and github actions for release system ... maybe an improvment issue ? :p

Comment (jycr) [reply pom.xml]: > By using the "[Maven Friendly Version](https://maven.apache.org/maven-ci-friendly.html)" guide, there is no more file modification needed in Git for a release.

I created an issue in green-code-initiative/ecoCode#208
We might migrate this feature to other repositories if it proves interesting

**Pull request merged**